compiler/front/optionsprocessor.nim

```diff
@@ -644,7 +644,7 @@ func testCompileOptionArg*(conf: ConfigRef; switch, arg: string): CompileOptArgC
   case switch.normalize
   of "gc":
     case arg.normalize
-    of "native": asResult conf.selectedGC == gcNative
+    of "default": asResult conf.selectedGC == gcDefault
```

Comment:
```suggestion
    of "default":
      let default =
        case conf.backend
        of backendC:             gcOrc
        of backendJs, backendVm: gcNative
        of backendInvalid:       unreachable() 
      asResult conf.selectedGC == default
```

The `testCompilationOptionArg` is used to implement the `compileOption("gc", x)` magic, so a `gcDefault` cannot reach here.

Having said that, I can currently not think of a situtation where querying whether the default GC is selected makes sense, so I'm not sure if allowing to test for "default" should be allowed.

Is there a use-case you have in mind?